Sources/FuturedHelpers/Helpers/CoordinatorSceneFlowProvider.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  CoordinatorSceneFlowProvider.swift
+//
+//  Created by Simon Sestak on 31/07/2024.
+//
+
+import SwiftUI
+import FuturedArchitecture
+
+/**
+ A typealias representing navigable destinations in a reusable scene flow.
+
+ # Notes: #
+ 1. If the scenes defined in the provider can continue with other scenes, define an end destination.
+    - This end destination should trigger a scene display outside of the scene provider.
+ 2. Other destinations will be used to display scenes defined within the scene provider.
+
+ # Example #
+ ```
+ protocol TemplateFlowDestination: CoordinatorSceneFlowDestination {
+    static var destination: Self { get }
+    static var end: Self { get }
+ }
+ ```
+ */
+public typealias CoordinatorSceneFlowDestination = Hashable & Identifiable & Equatable
+
+
+/**
+ A protocol providing an interface for reusable scene flow providers.
```

Comment:
What is a _reusable scene flow provider_? Is my suggestion correct?

```suggestion
 A protocol providing an interface for reusable scene flow providers. *Reusable scene flow provider* is part of a scene flow, which can be used as a part of more *Flow Coordinators*. The shared section of the flow is taken out of the *Flow Coordinator* and placed into a class conforming to `CoordinatorSceneFlowProvider`. 
```